KIPs/kip-82.md

```diff
@@ -0,0 +1,334 @@
+            shares[node.RewardAddr] = reward_amount
+
+    return (shares, remaining)
+
+# Unlike calc_deferred_reward, this function calculates the actual reward amounts
+# paid in this block. Used in klay_getReward RPC
+# Retruns a RewardSpec.
+def calc_actual_reward(header, config, staking_info):
+    spec = calc_deferred_reward(header, config, staking_info)
+
+    # If not DeferredTxFee, include block gas fee because they were ignored in
+    # calc_deferred_reward.
+    if not config.DeferredTxFee:
+        if header.number >= MAGMA_BLOCK_NUMBER:
+            block_fee += header.GasUsed * header.BaseFee
+        else:
+            block_fee += header.GasUsed * config.UnitPrice
+        spec.Proposer += block_fee
+        spec.Rewards[header.RewardBase] += block_fee
+
+    return spec
+```
+
+The update is expected to increase the amount of per-block state changes by number of GCs. The increase should be reasonable since the amount is significantly smaller than Klaytn's transaction processing capability.
+
+### Reward RPC
+
+A new JSON-RPC method is added to provide historic reward distribution details.
+
+- Name: `klay_getRewards`
+- Description: Returns allocation details of reward distribution at specified block. If the parameter is not set, returns a breakdown of reward distribution at the lastest block.
+- Parameters
+  1. `QUANTITY | TAG` - (optional) integer or hexadecimal block number, or the string "earlist" or "latest".
+- Returns
+  - `DATA`
+    - `source`
+      - `minted`: The amount minted
+      - `fee`: Total tx fee spent
+      - `burnt`: The amount burnt
+    - `output`
+      - `proposer`: The amount block proposer receives
+      - `stakers`: Total amount stakers receive
+      - `kgf`: The amount KGF receives
+      - `kir`: The amount KIR receives
+    - `rewards`: A mapping from reward recipient address to reward amount
+- Example
+  ```
+  // Request
+  curl -X POST -H "Content-Type: application/json" --data '{"jsonrpc":"2.0", "method":"klay_getRewards", "params":["0x5f5e100"],"id":1}' https://api.baobab.klaytn.net:8651
+  // Response
+  {
+    "jsonrpc":"2.0",
+    "id":1,
+    "result":{
+      "source": {
+        "minted": "9600000000000000000",
+        "fee": "4616950000000000",
+        "burnt": "2308475000000000",
+      },
+      "output": {
+        "proposer": "3264784881500000000",
+        "stakers": "0",
+        "kgf": "5185246576500000000",
+        "kir": "1152277017000000000"
+      },
+      "rewards": {
+        "0x99fb17d324fa0e07f23b49d09028ac0919414db6": "3264784881500000000",
+        "0x2bcf9d3e4a846015e7e3152a614c684de16f37c6": "5185246576500000000",
+        "0x716f89d9bc333286c79db4ebb05516897c8d208a": "1152277017000000000"
+      }
+    }
+  }
+  ```
+
+## Expected Effect
+The proposed GC reward mechanism is expected to produce the following changes:
+- GC members increase individual staking amount.
+- The higher number of KLAY holders receive profit from staking.
```

Comment:
What do you mean by "higher number of KLAY"? I rephrase it like below. Is that same meaning?
```suggestion
- KLAY holders receive profit based on the staking amount of KLAY.
```

Reply:
Can you rephrase @yeri-lee?